### PR TITLE
Type fixes, added params for quickstarts

### PR DESCRIPTION
**Pull request by radekkaluzik**
https://issues.redhat.com/browse/RHCLOUD-19169

### CODE REVIEW

**Review by Hyperkid123 — COMMENTED** [review]
@radekkaluzik can you also generate the spec base on these rules? its n the makefile jobs

File: cmd/spec/path.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ paths:
   /quickstarts/{id}:
     get:
       summary: Return a quickstarts by ID
+      parameters:
+      - $ref: '#/components/schemas/queryParams/Id'
```
Comment (Hyperkid123) [inline cmd/spec/path.yaml:32]: You have to do something similar for the `/helptopics/{name}:` endpoint.

**Comment by radekkaluzik** [comment]
> @radekkaluzik can you also generate the spec base on these rules? its n the makefile jobs

Done

File: cmd/spec/path.yaml (hunk 1 above)
Comment (radekkaluzik) [reply cmd/spec/path.yaml:32]: Fixed

File: cmd/spec/main.go [hunk 2]
```diff
@@ -46,6 +46,11 @@ func main() {
 		if name == "deletedAt" {
 			schema.Type = "string"
 			schema.Format = "date-time"
+			schema.OneOf = append(schema.OneOf, &openapi3.SchemaRef{Value: &openapi3.Schema{
```
Comment (Hyperkid123) [inline cmd/spec/main.go:49]: YOu have to do the same for the quickstart entity.

**Review by Hyperkid123 — CHANGES_REQUESTED** [review]

File: cmd/spec/main.go (hunk 2 above)
Comment (radekkaluzik) [reply cmd/spec/main.go:54]: ok, fixed

Comment (radekkaluzik) [reply cmd/spec/main.go:54]: I have also re-generated the specs

File: cmd/spec/path.yaml [hunk 3]
```diff
@@ -69,6 +71,8 @@ paths:
   /helptopics/{name}:
     get:
       summary: Return a help topics set by topic name
+      parameters:
+      - $ref: '#/components/schemas/queryParams/Name'
```
Comment (Hyperkid123) [inline cmd/spec/path.yaml:75]: You can't use the name type `Name`. That one is for query params. This one should be similar to the ID. It has to be in path and do not have any of the additional properties like explode, etc. It is also required as the ID

Comment (radekkaluzik) [reply cmd/spec/path.yaml]: Fixed

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**